js/components/scrubberBar.js

```diff
@@ -238,41 +239,56 @@ var ScrubberBar = React.createClass({
+
     if (this.props.controller.state.thumbnails && (this.state.scrubbingPlayheadX || this.lastScrubX || this.state.hoveringX)) {
       if (this.state.scrubbingPlayheadX) {
-        var hoverPosition = this.state.scrubbingPlayheadX;
-        var hoverTime = (this.state.scrubbingPlayheadX / this.state.scrubberBarWidth) * this.props.duration;
+        hoverPosition = this.state.scrubbingPlayheadX;
+        hoverTime = (this.state.scrubbingPlayheadX / this.state.scrubberBarWidth) * this.props.duration;
         playheadClassName += " oo-playhead-scrubbing";
-      }
-      else if (this.lastScrubX) {//to show thumbnail when clicking on playhead
-        var hoverPosition = this.props.currentPlayhead * this.state.scrubberBarWidth / this.props.duration;
-        var hoverTime = this.props.currentPlayhead;
+
+        thumbnailCarousel =
+          (<div className="oo-scrubber-carousel-container">
+           <ThumbnailCarousel
+           thumbnails={this.props.controller.state.thumbnails}
+           duration={this.props.duration}
+           hoverTime={hoverTime > 0 ? hoverTime : 0}
+           scrubberBarWidth={this.state.scrubberBarWidth}/>
+           </div>);
+      } else if (this.lastScrubX) {//to show thumbnail when clicking on playhead
+        hoverPosition = this.props.currentPlayhead * this.state.scrubberBarWidth / this.props.duration;
+        hoverTime = this.props.currentPlayhead;
         playheadClassName += " oo-playhead-scrubbing";
-      }
-      else if (this.state.hoveringX) {
-        var hoverPosition = this.state.hoveringX;
-        var hoverTime=(this.state.hoveringX / this.state.scrubberBarWidth) * this.props.duration;
-        var hoveredIndicatorStyle = {
-              width: Math.min((parseFloat(hoverTime) / parseFloat(this.props.duration)) * 100, 100) + "%",
-              backgroundColor: this.props.skinConfig.controlBar.scrubberBar.playedColor
-            };
+      } else if (this.state.hoveringX) {
+        hoverPosition = this.state.hoveringX;
+        hoverTime = (this.state.hoveringX / this.state.scrubberBarWidth) * this.props.duration;
+        hoveredIndicatorStyle = {
+          width: Math.min((parseFloat(hoverTime) / parseFloat(this.props.duration)) * 100, 100) + "%",
+          backgroundColor: this.props.skinConfig.controlBar.scrubberBar.playedColor
+        };
         scrubberBarClassName += " oo-scrubber-bar-hover";
         playheadClassName += " oo-playhead-hovering";
       }
-      thumbnailContainer =
-        (<div className="oo-scrubber-thumbnail-container">
-          <Thumbnail
-            thumbnails={this.props.controller.state.thumbnails}
-            hoverPosition={hoverPosition}
-            duration={this.props.duration}
-            hoverTime={hoverTime > 0 ? hoverTime : 0}
-            scrubberBarWidth={this.state.scrubberBarWidth}/>
-        </div>);
+      if (!thumbnailCarousel) {
+        thumbnailContainer =
+          (<div className="oo-scrubber-thumbnail-container">
```

Comment:
OK, for ThumbnailCarousel it is OK, but for Thumbnail it will become:  <div className="oo-scrubber-thumbnail-container oo-thumbnail" ref="thumbnail" style={thumbnailStyle}> - so it looks like we should merge classes oo-scrubber-thumbnail-container and oo-thumbnail as one of them becomes redundant, right?